arrow-core-data/src/main/kotlin/arrow/core/MapK.kt

```diff
@@ -13,8 +13,8 @@ data class MapK<K, out A>(private val map: Map<K, A>) : MapKOf<K, A>, Map<K, A>
   fun <B, Z> map2(fb: MapK<K, B>, f: (A, B) -> Z): MapK<K, Z> =
     if (fb.isEmpty()) emptyMap<K, Z>().k()
     else this.map.flatMap { (k, a) ->
-      fb.getOption(k).map { Tuple2(k, f(a, it)) }.k().asIterable()
-    }.k()
+      fb[k]?.let { Pair(k, f(a, it)) }.asIterable()
```

Comment:
nit: not a must, but I'd keep the Tuple2 here and below if possible for consistency

```suggestion
      fb[k]?.let { Tuple2(k, f(a, it)) }.asIterable()
```

Reply:
Copying message from other comment: @nomisRev suggested we switch to pair for this one. The impression I got is that Pair is slightly more efficient than Tuple2 in this case.